pkg/volume/host_path/host_path.go

```diff
@@ -103,6 +104,12 @@ func (plugin *hostPathPlugin) NewMounter(spec *volume.Spec, pod *v1.Pod, _ volum
 	if err != nil {
 		return nil, err
 	}
+
+	err = validation.ValidatePathNoBacksteps(hostPathVolumeSource.Path)
```

Comment:
@saad-ali It seems weird to me that validation would be in `SetUp` and there be no actual setup logic. At that point,`SetUp` a misnomer.